src/dmd/cpreprocess.d

```diff
@@ -0,0 +1,53 @@
+/**
+ * Run the C preprocessor on a C source file.
+ *
+ * Specification: C11
+ *
+ * Copyright:   Copyright (C) 2022 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 https://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 https://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/cpreprocess.d, _cpreprocess.d)
+ * Documentation:  https://dlang.org/phobos/dmd_cpreprocess.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/cpreprocess.d
+ */
+
+module dmd.cpreprocess;
+
+import core.stdc.stdio;
+import core.stdc.string;
+
+import dmd.astenums;
+import dmd.globals;
+import dmd.link;
+import dmd.errors;
+
+import dmd.common.outbuffer;
+
+import dmd.root.array;
+import dmd.root.filename;
+import dmd.root.rmem;
+import dmd.root.rootobject;
+import dmd.root.string;
+
+/***************************************
+ * Preprocess C file.
+ * Params:
+ *      csrcfile = C file to be preprocessed, with .c or .h extension
+ * Result:
+ *      filename of output
+ */
+FileName preprocess(FileName csrcfile)
+{
+    //printf("preprocess %s\n", csrcfile.toChars());
+    const name = FileName.name(csrcfile.toString());
+    const ext = FileName.ext(name);
+    assert(ext);
+    const ifilename = FileName.addExt(name[0 .. name.length - (ext.length + 1)], i_ext);
+    auto status = runPreprocessor("cpp", csrcfile.toString(), ifilename);
+    if (status)
+    {
+        error(Loc.initial, "C preprocess failed\n");
+        fatal();
```

Comment:
We should give a better error message here. If we can't extract the error message from `cpp` we should at least tell the user which file failed preprocessing (this was meant to be part of the previous review)

Reply:
Fair. Change that to `error(Loc.initial, "C preprocess failed for file %s\n", csrcfile.toChars());` then.